module/Category/src/Controller/CategoryController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Category\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Doctrine\ORM\EntityManager;
+use Category\Repository\CategoryRepository;
+
+class CategoryController extends AbstractActionController
+{
+    private $entityManager;
+
+    public function __construct($serviceManager)
+    {
+       $this->entityManager = $serviceManager->get(EntityManager::class);
+    }
+
+    public function indexAction()
+    {
+        $categories = new CategoryRepository($this->entityManager);
```

Comment:
Rather than performing logic inside the controller, you should create a service to perform this logic (and call the service methods from the controller). As the logic needed gets more complicated (in a real application) this keeps the business logic cleanly separated from the controller (separating concerns and making testing easier).